### PR TITLE
Fix DetailsRow's width not filling container by default

### PR DESCRIPTION
**Pull request by KevinTCoughlin**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

I noticed a rendering issue with `DetailsRow` in the `GroupedList` example. The `DetailsRow` of each group's list were not rendering with `width: 100%` to fill their container.

This was due to an in-line style overriding the `minWidth` set via `getStyles`:

https://github.com/OfficeDev/office-ui-fabric-react/blob/31c96451b1eee3e556fda7d156d19550a1c801f4/packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.base.tsx#L255

#### Screenshots

**Note:** I've added a `border: 1px solid black` to help show the width issue via screenshot.

**Before**

image

**After**

image

#### The Change

I've commented in-line on the relevant changes in this PR. The rest are prettier changes.

- Moved the conditional `minWidth` style calculation into `DetailsRow#getStyles` including the `viewport` prop as optional because the conditional requires it.
- Unrelated, but removed a redundant `css(classNames.root)` call on `DetailsRow`.
- Remainder are unrelated prettier changes...

#### Focus areas to test

- Render a `GroupedList` or look at the example on the local dev site at http://localhost:4322/#examples/groupedlist
- Hover over the rows to see the width more easily
- With this change, the hover style should go edge-to-edge.
- Compare with `master` where it wraps its contents.

I'll see if I can add this to our snapshots suite.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.base.tsx [hunk 1]
```diff
@@ -212,7 +195,8 @@ export class DetailsRowBase extends BaseComponent<IDetailsRowBaseProps, IDetails
       checkboxCellClassName,
       droppingClassName,
       className,
-      compact
+      compact,
+      viewport
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.base.tsx]: Added `viewport` as optional param via `IStyle`

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.base.tsx [hunk 2]
```diff
@@ -244,15 +226,14 @@ export class DetailsRowBase extends BaseComponent<IDetailsRowBaseProps, IDetails
         role="row"
         aria-label={ariaLabel}
         ariaDescribedBy={ariaDescribedBy}
-        className={css(classNames.root)}
+        className={classNames.root}
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.base.tsx]: Rm'ed redundant `css()`

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.styles.ts [hunk 3]
```diff
@@ -298,7 +299,7 @@ export const getStyles = (props: IDetailsRowStyleProps): IDetailsRowStyles => {
         background: colors.defaultBackgroundColor,
         color: colors.defaultMetaTextColor,
         display: 'inline-flex', // This ensures that the row always tries to consume is minimum width and does not compress.
-        minWidth: '100%',
+        minWidth: viewport ? viewport.width : '100%',
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.styles.ts]: Conditional has been moved from in-line to here and defaults to `100%` not `0`

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.types.ts [hunk 4]
```diff
@@ -184,36 +184,37 @@ export interface IDetailsRowProps extends IDetailsRowBaseProps {
   selectionMode: SelectionMode;
 }
 
-export type IDetailsRowStyleProps = Required<Pick<IDetailsRowProps, 'theme'>> & {
-  /** Whether the row is selected  */
-  isSelected?: boolean;
+export type IDetailsRowStyleProps = Required<Pick<IDetailsRowProps, 'theme'>> &
+  Pick<IDetailsRowProps, 'viewport'> & {
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.types.ts]: Picking optional `viewport` prop for `IStyle`

**Comment by KevinTCoughlin** [comment]
Worth calling out that this only happens on the `GroupedList` page locally not on https://developer.microsoft.com/en-us/fabric#/components/groupedlist even tho `min-width: 0`.

I tried to get context of the `in-line` style change via git history but it ends at css-in-js conversion.

**Comment by KevinTCoughlin** [comment]
@dzearing this fixes the wrap width behavior on the `GroupedList` examples page you noticed. Please take a look when you can.

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.base.tsx (hunk 2 above)
Comment (JasonGore) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.base.tsx]: Hmm, I question the need for the one on **new** line 274 too. And I'm curious if the rest should eventually be removed with their logic moved to `DetailsRow.styles.ts`?

Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.base.tsx]: Yea I think so, this is in a half state. Let me evaluate this offline and raise a PR separately for the remainder.

**Comment by KevinTCoughlin** [comment]
Thanks for reviewing this @natalieethell & @JasonGore. I can't merge this as-is because it flips the problem which will surface on our PROD site...

image

Glad I checked locally. Let me take some time today to try to resolve for both cases...

**Comment by KevinTCoughlin** [comment]
Will re-open shortly with a fix for both website contexts.

**Pull request closed without merging**